content/Projects/Observability/kube-rbac-proxy.md

```diff
@@ -0,0 +1,136 @@
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: kube-rbac-proxy
+type: Opaque
+data:
+  tls.crt: ...
+  tls.key: ...
+  ca.crt: ...
+---
+apiVersion: apps/v1
+kind: Deployment
+...
+spec:
+  template:
+    spec:
+      containers:
+      - name: kube-rbac-proxy
+        image: quay.io/brancz/kube-rbac-proxy:v0.8.0
+        args:
+        - "--tls-cert-file=/etc/tls/private/tls.crt"
+        - "--tls-private-key-file=/etc/tls/private/tls.key"
+        - "--client-ca-file=/etc/tls/client/client-ca.crt"
+        ...
+        volumeMounts:
+        - name: secret-volume
+          mountPath: /etc/tls
+      volumes:
+      - name: secret-volume
+        secret:
+          secretName: kube-rbac-proxy
+```
+
+CMO specifies the aforementioned CA certificate in the [metrics-client-ca](https://github.com/openshift/cluster-monitoring-operator/blob/6795f509e77cce6d24e5a3e371a432ca22e1a8e7/assets/cluster-monitoring-operator/metrics-client-ca.yaml) ConfigMap which is used to define client certificates for every `kube-rbac-proxy` container that's safeguarding a component. The component's `Service` endpoints are secured using the generated TLS `Secret` annotating it with the `service.beta.openshift.io/serving-cert-secret-name`. Internally, this requests the [`service-ca`](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.1/html/authentication/configuring-certificates#understanding-service-serving_service-serving-certificate) controller to generate a `Secret` containing a certificate and key pair for the `${service.name}.${service.namespace}.svc`. These TLS manifests are then used in various component `ServiceMonitors` to define their TLS configurations, and within CMO to ensure a "mutual" acknowledgement between the two.
+
+### Securing API endpoints
+
+- **Static authorization**: Static authorization involves configuring `kube-rbac-proxy` to allow access to certain resources or non-resources which are evaluated against the `Role` or `ClusterRole` RBAC permissions the user or the service account has. The example below demonstrates how this can be employed to give access to a known `ServiceAccount` to the `/metrics` endpoint. `/metrics` endpoints exposed by various monitoring components are protected this way. Note that after the initial user or service account authentication, the request is matched against a comma-separated list of paths, as defined by the [`--allow-path`](https://github.com/brancz/kube-rbac-proxy/blob/067e14a2d1ecdfe8c18da6b0a0507cd4684e2c1c/cmd/kube-rbac-proxy/app/options/options.go#L83) flag, [like so](https://github.com/openshift/cluster-monitoring-operator/blob/4e8efd9864bff3ff46499e86fef8bba1e0178f54/assets/alertmanager/alertmanager.yaml#L96).
```

Comment:
IIRC /metrics is the only case where we use static authorization. It would be good having a dedicated section and linking to the "Collecting metrics" page.

```
### Securing /metrics endpoint

<link to https://rhobs-handbook.netlify.app/products/openshiftmonitoring/collecting_metrics.md/#kube-rbac-proxy-sidecar>

### Securing API endpoints

...
```

Reply:
WDYT about my suggestion above?
Now that #62 has merged, we link to it for all the details about how the client cert/key for metrics collection is managed.

Reply:
I added the initial suggestion here: https://github.com/rhobs/handbook/pull/59/commits/5ab97a67664f8cd33868d196854ac2ee5c90bde0#diff-5a9d05e906a90f4d79960b53e1aaa1e69220d15b150527c2da2c863f1aa1657aR101. I've updated the link to: https://github.com/rhobs/handbook/blob/main/content/Products/OpenshiftMonitoring/collecting_metrics.md#exposing-metrics-for-prometheus.